crate/cli/src/actions/shared/import_key.rs

```diff
@@ -100,37 +102,72 @@ pub struct ImportKeyAction {
     /// To specify multiple tags, use the option multiple times.
     #[clap(long = "tag", short = 't', value_name = "TAG")]
     tags: Vec<String>,
+
+    /// For what operations should the key be used.
+    #[clap(long = "key-usage")]
+    key_usage: Option<Vec<KeyUsage>>,
 }
 
 impl ImportKeyAction {
     pub async fn run(&self, kms_rest_client: &KmsClient) -> Result<(), CliError> {
+        let cryptographic_usage_mask = build_usage_mask_from_key_usage(&self.key_usage);
         // read the key file
         let bytes = Zeroizing::from(read_bytes_from_file(&self.key_file)?);
         let object = match &self.key_format {
-            ImportKeyFormat::JsonTtlv => read_object_from_json_ttlv_bytes(&bytes)?,
-            ImportKeyFormat::Pem => read_key_from_pem(&bytes)?,
-            ImportKeyFormat::Sec1 => {
-                build_private_key_from_der_bytes(KeyFormatType::ECPrivateKey, bytes)
-            }
-            ImportKeyFormat::Pkcs1Priv => {
-                build_private_key_from_der_bytes(KeyFormatType::PKCS1, bytes)
-            }
-            ImportKeyFormat::Pkcs1Pub => {
-                build_public_key_from_der_bytes(KeyFormatType::PKCS1, bytes)
-            }
-            ImportKeyFormat::Pkcs8 => build_private_key_from_der_bytes(KeyFormatType::PKCS8, bytes),
-            ImportKeyFormat::Spki => build_public_key_from_der_bytes(KeyFormatType::PKCS8, bytes),
-            ImportKeyFormat::Aes => {
-                build_symmetric_key_from_bytes(CryptographicAlgorithm::AES, bytes)
-            }
-            ImportKeyFormat::Chacha20 => {
-                build_symmetric_key_from_bytes(CryptographicAlgorithm::ChaCha20, bytes)
+            ImportKeyFormat::JsonTtlv => {
+                let mut obj = read_object_from_json_ttlv_bytes(&bytes)?;
+                obj.attributes_mut()?
+                    .set_cryptographic_usage_mask(cryptographic_usage_mask);
+                obj
```

Comment:
I don't know. I agree this would be better but in the meantime, I feel like setting the cryptographic usage mask directly when constructing the object (inside the functions) would be somewhat better ?

Reply:
Yes, I agree, setting the mask afterwards somehow differs from setting it upon creation. Maybe it is easier to prove all keys created have an associated mask if it is not possible to create a key without setting it.

In that case, at least enforce symmetry by creating your TTLV with the mask :)